Makefile

```diff
@@ -18,8 +18,22 @@ GENERATED_FILES := $(UNIT_FILES)
 UNIT_FILES += kata-containers.target
 endif
 
+VERSION := $(shell grep -v ^\# ./VERSION)
+COMMIT_NO := $(shell git rev-parse HEAD 2> /dev/null || true)
+COMMIT := $(if $(shell git status --porcelain --untracked-files=no),${COMMIT_NO}-dirty,${COMMIT_NO})
+
+DESCRIBE := $(shell git describe 2> /dev/null || true)
+DESCRIBE_DIRTY := $(if $(shell git status --porcelain --untracked-files=no 2> /dev/null),${DESCRIBE}-dirty,${DESCRIBE})
+GIT_COMMIT := $(shell git rev-parse HEAD 2>/dev/null)
+ifneq ($(GIT_COMMIT),)
+VERSION_COMMIT := $(VERSION)-$(GIT_COMMIT)
+endif
+ifneq ($(DESCRIBE_DIRTY),)
```

Comment:
I think it would be better to just rely  on `VERSION` and  `COMMIT`.  Using git describe will lead to build an agent with a version value based in the last tag a not in `./VERSION` file.